### PR TITLE
[ASImageNode]fix incorrect backing size calculation

### LINKED ISSUE #1052: [ASImageNode]Incorrect calculation for backing size calculation
In demo project and screenshot, I create 2 simple `ASImageNode` with `UIViewContentModeScaleAspectFit`.
However, change the height of node will make image "blurred".

I also find that #836 has mentioned this issue too.I read the code about `ASCroppedImageBackingSizeAndDrawRectInBounds`, and I think there is something wrong with image draw rect calculation.

[Sample Project](https://github.com/junjielu/ImageNodeTest)

preview
@nguyenhuy @appleguy

### PR DESCRIPTION
**Pull request by junjielu**
Try to fix #1052 .

In `ASCroppedImageBackingSizeAndDrawRectInBounds`, we want to calculate size which is minimum destination size for backing. By giving bounds ratio and image size, it's a **reverse** process to calculate the appropriate destination size and previous coder may ignore this.

### CODE REVIEW

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/1b9995952a1397348bb0c944be050419/raw/9c1d2bebd24e33be572e4d4ca13584d78d7370b0/log.txt)

**Comment by junjielu** [comment]
In `ASCroppedImageBackingSizeAndDrawRectInBounds `, we just want to calculate accurate backing size and draw rect.

If `contentMode == .scaleAspectFit`, we expect backing size can "contain" image.And if `contentMode == .scaleAspectFill`, we expect backign size can be "contained" inside image.

For example, if we have a 200x400(pixels) view and a 100x100(pixels) image, the backing size and draw rect calculation are just like pic shows below.

untitled diagram 3

By given bounds ratio and image size, we just want to calculate `minimumDestinationSize`, and if result is small than bounds size, the `minimumDestinationSize` will be backing size.

However, I found that functions `_ASSizeFillWithAspectRatio` and `_ASSizeFitWithAspectRatio` is designed to receive content ratio and destionation size to calculate scaled size for content.So, the `minimumDestinationSize` calculation is a **reverse** process for these functions which means we need use 
- `_ASSizeFillWithAspectRatio` to calculate `minimumDestinationSize` in scaleAspectFit content mode
- `_ASSizeFitWithAspectRatio` to calculate `minimumDestinationSize` in scaleAspectFill content mode

**Comment by junjielu** [comment]
@adlai-holler @nguyenhuy @maicki @appleguy  This issue would make backing size too big for scale aspect fill content mode and too small for scale aspect fit content mode. Please help review if this fix works :)

File: Source/Private/ASImageNode+CGExtras.m [hunk 1]
```diff
@@ -53,26 +53,27 @@ void ASCroppedImageBackingSizeAndDrawRectInBounds(CGSize sourceImageSize,
...
   BOOL cropToRectDimensions = !CGRectIsEmpty(cropRect);
 
+  // Given image size and container ratio, calculate minimum container size for the image under different contentMode
   if (cropToRectDimensions) {
-    scaledSizeForImage = CGSizeMake(boundsSize.width / cropRect.size.width, boundsSize.height / cropRect.size.height);
+    minimumDestinationSize = CGSizeMake(boundsSize.width / cropRect.size.width, boundsSize.height / cropRect.size.height);
   } else {
     if (contentMode == UIViewContentModeScaleAspectFill)
-      scaledSizeForImage = _ASSizeFillWithAspectRatio(boundsAspectRatio, sourceImageSize);
+      minimumDestinationSize = _ASSizeFitWithAspectRatio(boundsAspectRatio, sourceImageSize);
```
Comment (wiseoldduck) [inline Source/Private/ASImageNode+CGExtras.m]: This is of course the meat of the fix

**Comment by wiseoldduck** [comment]
LGTM (minus conflicts)

**Comment by junjielu** [comment]
Thanks for your review~ Conflicts resolved.

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR modifies CHANGELOG.md, which is a generated file. Add #changelog to the title to suppress this warning.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by wiseoldduck** [comment]
Oh yeah the CHANGELOG happens automatically now, so probably revert that part :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 **Merged pull requests:**
 
+- [ASImageNode] Fix incorrect backing size calculation. [\#1189](https://github.com/TextureGroup/Texture/pull/1189) ([Junjie Lu](https://github.com/junjielu))
```
Comment (wiseoldduck) [inline CHANGELOG.md]:
```suggestion
```

**Comment by junjielu** [comment]
Sure, of course.

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please ensure license is correct for ASImageNodeBackingSizeTests.mm:</p>
<pre><code>//
//  ASImageNodeBackingSizeTests.mm
//  Texture
//
//  Copyright (c) Pinterest, Inc.  All rights reserved.
//  Licensed under Apache 2.0: http://www.apache.org/licenses/LICENSE-2.0
</code></</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**